src/dimension/Dimension.js

```diff
@@ -0,0 +1,488 @@
+        const { _dimensionIndex } = this
+        if (append) {
+            this._scopedFilters[_dimensionIndex] += "NOT( " + this.formILikeExpression(value) + ")"
+        } else {
+            this._scopedFilters[_dimensionIndex] = "NOT( " + this.formILikeExpression(value) + ")"
+        }
+        return this
+    }
+    filterIsNotNull(append) {
+        const { _dimensionIndex, _expression } = this
+        if (append) {
+            this._scopedFilters[_dimensionIndex] += `${_expression} IS NOT NULL`
+        } else {
+            this._scopedFilters[_dimensionIndex] = `${_expression} IS NOT NULL`
+        }
+        return this
+    }
+    filterIsNull(append) {
+        const { _dimensionIndex, _expression } = this
+        if (append) {
+            this._scopedFilters[_dimensionIndex] += `${_expression} IS NULL`
+        } else {
+            this._scopedFilters[_dimensionIndex] = `${_expression} IS NULL`
+        }
+        return this
+    }
+    filterRange(range, append = false, resetRange, inverseFilters, binParams, isRelative) {
+        // console.log('Dimension.filterRange()')
+        const { _dimensionIndex, _dimArray } = this,
+            isArray       = Array.isArray(range[0]) // TODO semi-risky index
+        let subExpression = ""
+
+        if (!isArray) {
+            range = [range]
+        }
+        this._filterVal = range
+
+        for (let e = 0; e < range.length; e++) {
+            if (resetRange === true) {
+                this._rangeFilters[e] = range[e]
+            }
+            if (e > 0) {
+                subExpression += " AND "
+            }
+
+            let typedRange = [
+                formatFilterValue(range[e][0], true),
+                formatFilterValue(range[e][1], true)
+            ]
+            if (isRelative) {
+                typedRange = [
+                    this.formatRelativeValue(typedRange[0]),
+                    this.formatRelativeValue(typedRange[1])
+                ]
+            }
+            if (binParams && binParams[e] && binParams[e].extract) {
+                const dimension = "extract(" + binParams[e].timeBin + " from " + uncast(_dimArray[e]) +  ")"
+                subExpression += dimension + " >= " + typedRange[0] + " AND " + dimension + " <= " + typedRange[1]
+            } else {
+                subExpression += _dimArray[e] + " >= " + typedRange[0] + " AND " + _dimArray[e] + " <= " + typedRange[1]
+            }
+        }
+        if (inverseFilters) {
+            subExpression = "NOT(" + subExpression + ")"
+        }
+        if (append) {
+            this._scopedFilters[_dimensionIndex] += "(" + subExpression + ")"
+        } else {
+            this._scopedFilters[_dimensionIndex] = "(" + subExpression + ")"
+        }
+        return this
+    }
+    formatRelativeValue(val) {
+        if (val.now) {
+            return "NOW()" // todo - there might be subtle bugs in Now(), depending on when it's called & evaluated
```

Comment:
If we think there's a bug here let's fix it. otherwise if a bug arises they'll fix it then.